logbook-json/src/main/java/org/zalando/logbook/json/JsonPathBodyFilters.java

```diff
@@ -0,0 +1,91 @@
+package org.zalando.logbook.json;
+
+import com.jayway.jsonpath.Configuration;
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+import com.jayway.jsonpath.Option;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import org.zalando.logbook.BodyFilter;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class JsonPathBodyFilters {
+
+    private static final Configuration conf = Configuration.builder().options(Option.AS_PATH_LIST).build();
+
+    @RequiredArgsConstructor
+    public static class JsonPathBodyFilterBuilder {
+        private final JsonPath jsonPath;
+
+        public BodyFilter delete() {
+            return (JsonBodyFilter) documentContext -> documentContext.delete(jsonPath).jsonString();
+        }
+
+        public BodyFilter replace(String replacement) {
+            return replaceInternal(replacement);
+        }
+
+        public BodyFilter replace(Number replacement) {
+            return replaceInternal(replacement);
+        }
+
+        public BodyFilter replace(Boolean replacement) {
+            return replaceInternal(replacement);
+        }
+
+        public BodyFilter replace(String matchRegex, String replacementRegex) {
+            Pattern pattern = Pattern.compile(matchRegex);
+            return (JsonBodyFilter) documentContext -> {
+                List<String> paths = documentContext.read(jsonPath);
+                Object jsonObject = documentContext.json();
+                paths.forEach(path -> {
+                    Object obj = JsonPath.compile(path).read(jsonObject);
+                    replaceWithRegex(obj, pattern, replacementRegex)
+                            .ifPresent(s -> documentContext.set(path, s));
+                });
+                return documentContext.jsonString();
+            };
+        }
+
+        private BodyFilter replaceInternal(Object replacement) {
+            return (JsonBodyFilter) documentContext -> {
+                List<String> obj = documentContext.read(jsonPath);
+                obj.forEach(path -> documentContext.set(path, replacement));
+                documentContext.set(jsonPath, replacement);
+                return documentContext.jsonString();
+            };
+        }
+
+        private Optional<String> replaceWithRegex(Object obj, Pattern pattern, String replacementRegex) {
+            String stringRepresentation = obj.toString();
+            Matcher matcher = pattern.matcher(stringRepresentation);
+            if (matcher.find()) {
+                return Optional.of(matcher.replaceAll(replacementRegex));
+            }
+            return Optional.empty();
+        }
+
+    }
```

Comment:
I believe this conditional/optional handling is not really necessary and could be simplified:

```suggestion
        private String replace(Object node, Pattern pattern, String replacement) {
            Matcher matcher = pattern.matcher(node.toString());
            return matcher.replaceAll(replacement));
        }

    }
```

Reply:
That logics were implemented to make replacement a bit more fail-safe. For example, if we have some number, which didn't match regex, it will be converted into String, when using your variant. If we have object/array - it will behave in the same way. 
If it doesn't matter - i can update this logics :)

Reply:
Ah, I totally missed that. Good point. Better leave it like it is then. Is that covered by a test? I guess so.